src/streamlink/plugins/ltv_lsm_lv.py

```diff
@@ -1,45 +1,83 @@
+        validate.get("source"), validate.get("embed_code"),
+        validate.parse_html(),
+        validate.xml_xpath_string(".//iframe/@src"),
+    )
+
+    player_apicall_schema = validate.Schema(
+        validate.transform(_json_data_re.search),
+        validate.any(
+            None,
+            validate.all(
+                validate.get(1),
+                validate.transform(lambda s: s.replace("'", '"')),
+                validate.transform(lambda s: re.sub(r",\s*\}", "}", s, flags=re.DOTALL)),
+                validate.parse_json(),
+                {"channel": validate.text},
+                validate.get("channel")
+            )
+        )
+    )
+
+    sources_schema = validate.Schema({
+        "source": {
+            "sources": validate.all(
+                [{"type": validate.text, "src": validate.url()}],
+                validate.filter(lambda src: src["type"] == "application/x-mpegURL"),
+                validate.map(lambda src: src.get("src"))
+            ),
+        }},
+        validate.get('source'),
+        validate.get('sources')
+    )
+
     def _get_streams(self):
         self.session.http.headers.update({"Referer": self.url})
 
-        iframe_url = None
-        res = self.session.http.get(self.url)
-        for iframe in itertags(res.text, "iframe"):
-            if "embed.lsm.lv" in iframe.attributes.get("src"):
-                iframe_url = iframe.attributes.get("src")
-                break
-
-        if not iframe_url:
-            log.error("Could not find player iframe")
-            return
-
-        log.debug("Found iframe: {0}".format(iframe_url))
-        res = self.session.http.get(iframe_url)
-        for source in itertags(res.text, "source"):
-            if source.attributes.get("src"):
-                stream_url = source.attributes.get("src")
-                url_path = urlparse(stream_url).path
-                if url_path.endswith(".m3u8"):
-                    yield from HLSStream.parse_variant_playlist(self.session, stream_url).items()
-                else:
-                    log.debug("Not used URL path: {0}".format(url_path))
+        iframe_url = self.session.http.get(self.url, schema=self.main_page_schema)
+        log.debug(f"Found embed iframe: {iframe_url}")
+        player_iframe_url = self.session.http.get(iframe_url, schema=self.embed_code_schema)
+        log.debug(f"Found player iframe: {player_iframe_url}")
+        channel_id = self.session.http.get(player_iframe_url, schema=self.player_apicall_schema)
+        log.debug(f"Found channel ID: {channel_id}")
+
+        stream_sources = self.session.http.post(self.API_URL.format(channel_id=channel_id),
+                                                data=dict(refer="ltv.lsm.lv",
+                                                          playertype="regular",
+                                                          protocol="hls"))
+        for surl in self.session.http.json(stream_sources, schema=self.sources_schema):
```

Comment:
Instead of calling `self.session.http.json`, you could add `validate.parse_json()` to the used schema, which fits the style of the other schemas and avoids the secondary method call on the http session.

Reply:
I felt this was more readable.